src/compiler/compiler.rs

```diff
@@ -921,129 +944,176 @@ where
+                        match proxy.resolve_proxied_executable(creator1, cwd, env).await {
+                            Ok((resolved_path, _time)) => {
+                                trace!("Resolved path with rustup proxy {:?}", &resolved_path);
+                                Ok((Some(proxy), resolved_path))
+                            }
+                            Err(e) => {
+                                trace!("Could not resolve compiler with rustup proxy: {}", e);
+                                Ok((None, rustc_executable))
                             }
-                        }
-                        Ok(None) => {
-                            trace!("Did not find rustup");
-                            Ok((None, rustc_executable))
-                        }
-                        Err(e) => {
-                            trace!("Did not find rustup due to {}, compiling without proxy", e);
-                            Ok((None, rustc_executable))
                         }
                     }
-                });
+                    Ok(None) => {
+                        trace!("Did not find rustup");
+                        Ok((None, rustc_executable))
+                    }
+                    Err(e) => {
+                        trace!("Did not find rustup due to {}, compiling without proxy", e);
+                        Ok((None, rustc_executable))
+                    }
+                }
+            });
 
-                let (proxy, resolved_rustc) = res
-                    .await
-                    .map(|(proxy, resolved_compiler_executable)| {
-                        (
-                            proxy
-                                .map(Box::new)
-                                .map(|x: Box<RustupProxy>| x as Box<dyn CompilerProxy<T>>),
-                            resolved_compiler_executable,
-                        )
-                    })
-                    .unwrap_or_else(|_e| {
-                        trace!("Compiling rust without proxy");
-                        (None, rustc_executable2)
-                    });
-
-                debug!("Using rustc at path: {resolved_rustc:?}");
-
-                Rust::new(
-                    creator,
-                    resolved_rustc,
-                    env,
-                    &rustc_verbose_version,
-                    dist_archive,
-                    pool,
-                )
+            let (proxy, resolved_rustc) = res
                 .await
-                .map(|c| {
+                .map(|(proxy, resolved_compiler_executable)| {
                     (
-                        Box::new(c) as Box<dyn Compiler<T>>,
-                        proxy as Option<Box<dyn CompilerProxy<T>>>,
+                        proxy
+                            .map(Box::new)
+                            .map(|x: Box<RustupProxy>| x as Box<dyn CompilerProxy<T>>),
+                        resolved_compiler_executable,
```

Comment:
This became less readable than before

Reply:
Are you doing per commit review as I cannot find this comment on full view?

Reply:
I am doing a full diff view, and the changeset exists.